reconcile/utils/ldap_client.py

```diff
@@ -34,14 +36,43 @@ def get_users(self, uids: Iterable[str]) -> set[str]:
         )
         return set(r["attributes"]["uid"][0] for r in results)
 
+    def get_rover_groups(self, groups: set[str]) -> dict[str, set[str]]:
+        if not groups:
+            return {}
+        rover_group_dn_mapping = {
+            f"cn={cn},ou=adhoc,ou=managedGroups,dc=redhat,dc=com": cn for cn in groups
+        }
+        rover_group_dns = set(rover_group_dn_mapping.keys())
+        filter = f"(|{''.join([f'(memberOf={dn})' for dn in rover_group_dns])})"
+
+        _, _, users, _ = self.connection.search(
+            self.base_dn,
+            filter,
+            attributes=["uid", "memberOf"],
+        )
+        groups_and_members: dict[str, set[str]] = defaultdict(set[str])
+        for u in users:
+            uid = u["attributes"]["uid"][0]
+            for group in set(u["attributes"]["memberOf"]).intersection(rover_group_dns):
+                groups_and_members[rover_group_dn_mapping[group]].add(uid)
```

Comment:
nit: use both `rover_group_dns` and `rover_group_dn_mapping` together plus nested access is a little hard understand. An easier to understand version can be:

```python
            member_groups = (
                rover_group_dn_mapping[g]
                for g in u["attributes"]["memberOf"]
                if g in rover_group_dn_mapping
            )
            for group in member_groups:
                groups_and_members[group].add(uid)
```